specification/2.0/appendix/performance-metrics/README.md

```diff
@@ -0,0 +1,79 @@
+|[nodeCount]      | Integer     | Complexity  |Scene     |Max nodecount in scene (add upp all nodes in a scene)           |  
+|[primitiveCount] | Integer     | Complexity  |Scene     |Total number of referenced primitives (per scene).  This figure is the un-batched number of primitives, engines may optimize if primitives and meshes share textures. |  
+|[textures]       | Integer     | Complexity  |Scene     |Flags specifying presence of materials and it's texture usage, this is the aggregated max usage. BASECOLOR, METALLICROUGHNESS, NORMAL, OCCLUSION, EMISSIVE, (SPECULARGLOSS) |  
+|[attributes]     | Integer     | Memory      | Asset    | The total memory footprint, in bytes, for the models attribute buffer storage |  
+|[textureSizes]  |Dimension    | Memory      | Asset    | The width and height of defined textures in the Asset. |  
+
+Dimension is an Integer[2] containing width and height  
+
+
+# Scene #  
+
+Metrics that describe properties of a scene.  
+glTF models may contain multiple scenes, only one scene will be visible at a time.  
+To accomodate this the following metrics are calculated on a per-scene basis.  
+It is possible to reference different meshes (nodes) using different scenes, this makes it possible to have the same texture assets but reference different primitives that use alternative materials (textures).  
+Ie one model could have 3 different scenes, where the same accessors (position, uv) are used but different primitives that reference materials with varying number of texture channels.  
+
+[nodeCount]  
+This represent the number of traversed nodes in a scene.  
+This is calculated by traversing nodes in each scene (depth or breadth first does not matter) - for each node increase the nodecount.  
+
+[vertexCount]  
+This represents the number of drawn vertices for each scene in the model.  
+This is calculated by traversing the nodes in each scene.  
+For each mesh use the POSITION attribute in each primitive, get the Accessor and add up the count field.  
+
+[primitiveCount]  
+This represents the number of primitives for each scene  
+Each primitive references a material with 0 or more texture sources, has attributes and accessors.  
+
+
+[textures]  
+This value represents the texture sources that are used by a scene.  
+This is the max value from the most 'complex' primitive that will be referenced by a Node in the scene.  
+The goal of this metric is to provide a worst case texture usage where texturecount and complexity is known.  
+Ie it is known if a texture is BASECOLOR or if part of PBR such as METALICROUGHNESS and now many texture sources that are used.  
+
+[attributes]  
+This value represent the total attribute buffer usage of the model. This value is calculated by adding up the size, in bytes, of all buffer objects in the Asset.  
+This may give an indication of the runtime memory footprint of the buffers needed for the model.  
```

Comment:
Given the goal is to break down usage in terms of data, it would be more useful to split the attributes data further, I would say.    We have the attributes POSITION, NORMAL, TANGENT, TEXCOORD_0, TEXCOORD_1, COLOR_0, JOINTS_0, and WEIGHTS_0.  vertexCount is already handling the POSITION attribute.

The normals/tangents/texCoords/colors are very useful data to know, independent on its own.  A system can handle glTF assets in a lot of different ways such as point cloud or wireframe display (only using POSITION attribute), or shaded/textured display (using NORMAL, TANGENT attributes) or only with color attribute, or one set of texture coordinates (at a given time) etc.   So, having the breakdown would be very helpful.
What do you think of the metrics breakdown of these attributes per scene?

Reply:
I understand your point - however I think the same information could more or less be extrapolated from texture usage - and also that the overhead is very unlikely to be in the vertex stage.
I think that the vertexCount metric is more to give an indication of number of triangles (4000 vs 4 000 000).

I can see how the primitive mode value could perhaps be of interest:
>POINTS
LINES
LINE_LOOP
LINE_STRIP
TRIANGLES
TRIANGLE_STRIP
TRIANGLE_FAN
 
@raviramachandra What's your input on this?

Reply:
@rsahlin I agree with your analysis. The direct impact of these (apart from the obvious memory utilization) attributes on runtime is not obvious or in some cases like texcoords, can be extrapolated from the current information available. For now we can skip this info.

Another interesting point that came up during our internal discussion here was this: 
What if there is an asset which has very high vertex count, but it's overall screen space is very small. For example: say a 400000 vertex asset which may take very small screen space say : 5 cubic cm. I believe GPUs do some kind optimization and may save some compute during pixel shading or rasterization stages ?  Would you also recommend adding something like an static bounding box as additional metric ? Something similar to what FB has in their extension ?
https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Vendor/FB_geometry_metadata/README.md

Reply:
Yes, I think the screen space dimension somehow needs to be known for the usecase.
Perhaps the client can send this with the request to the backend?

So the client would ask the backend for an asset to be used for usecase A taking up X by Y pixels screen space.
Do you think this could work?

Apart from that I am not sure how we could incorporate it into these metrics.

Reply:
Added a section in the beginning that mention screen coverage as not being part of these metrics

Reply:
After reading through the Spatial Favicons background of this PR, and syncing with Ravi, I can now see the use-case for this metric better.  Also, @donmccurdy 's reference to W3C's responsive images is a good parallel (https://github.com/KhronosGroup/glTF/issues/1743).  Given that, my earlier broken down data would not fit those goals.
It made me thinking more on the screen coverage, and I would try to summarize my thoughts separately.